extensions/common/sql/sql-bootstrapper/src/main/java/org/eclipse/edc/sql/bootstrapper/SqlSchemaBootstrapperImpl.java

```diff
@@ -15,74 +15,33 @@
 public class SqlSchemaBootstrapperImpl implements SqlSchemaBootstrapper {
-    private final TransactionContext transactionContext;
-    private final QueryExecutor queryExecutor;
-    private final List<QueuedStatementRecord> statements = new ArrayList<>();
-    private final DataSourceRegistry dataSourceRegistry;
-    private final Monitor monitor;
-
-    public SqlSchemaBootstrapperImpl(TransactionContext transactionContext, QueryExecutor queryExecutor, DataSourceRegistry dataSourceRegistry, Monitor monitor) {
-        this.transactionContext = transactionContext;
-        this.queryExecutor = queryExecutor;
-        this.dataSourceRegistry = dataSourceRegistry;
-        this.monitor = monitor;
-    }
 
+    private final List<QueuedStatementRecord> statements = new ArrayList<>();
 
     @Override
     public void addStatementFromResource(String datasourceName, String resourceName, ClassLoader classLoader) {
-        try (var sqlStream = classLoader.getResourceAsStream(resourceName)) {
-            var sql = new Scanner(Objects.requireNonNull(sqlStream)).useDelimiter("\\A").next();
+        try (var sqlStream = classLoader.getResourceAsStream(resourceName); var scanner = new Scanner(Objects.requireNonNull(sqlStream)).useDelimiter("\\A")) {
+            var sql = scanner.next();
             statements.add(new QueuedStatementRecord(resourceName, datasourceName, sql));
         } catch (IOException e) {
             throw new EdcException(e);
         }
     }
 
-    public Result<Void> executeSql() {
-        monitor.debug("Running DML statements: [%s]".formatted(statements.stream().map(QueuedStatementRecord::name).collect(Collectors.joining(", "))));
-        return transactionContext.execute(() -> {
-            Stream<Result<Void>> objectStream = statements.stream().map(statement -> {
-                var connectionResult = getConnection(statement.datasourceName);
-                return connectionResult.compose(connection -> {
-                    queryExecutor.execute(connection, statement.sql);
-                    return success();
-                });
-            });
-            return objectStream.reduce(Result::merge).orElse(Result.success());
-        });
-    }
-
-    public Result<Connection> getConnection(String datasourceName) {
-        try {
-            var resolve = dataSourceRegistry.resolve(datasourceName);
-            return resolve != null ? success(resolve.getConnection()) :
-                    failure("No datasource found with name '%s'".formatted(datasourceName));
-        } catch (SQLException e) {
-            return failure(e.getMessage());
-        }
-    }
-
-    private record QueuedStatementRecord(String name, String datasourceName, String sql) {
+    @Override
+    public Map<String, List<String>> getStatements() {
+        var map = new HashMap<String, List<String>>();
+        statements.forEach(qsr -> map.computeIfAbsent(qsr.datasourceName(), s -> new ArrayList<>()).add(qsr.sql()));
+        return map;
```

Comment:
nit: this could be replaced by
```java
return statements.stream().collect(groupingBy(QueuedStatementRecord::datasourceName, mapping(QueuedStatementRecord::sql, toList())));
```